app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt

```diff
@@ -0,0 +1,418 @@
+
+private val creditCardPaymentArguments = listOf(
+  navArgument(CREDIT_CARD_PAYMENT_ID_ARG) {
+    type = NavType.StringType
+    nullable = false
+  },
+  navArgument(IS_PRE_SELECTED) {
+    type = NavType.BoolType
+    defaultValue = false
+    nullable = false
+  }
+)
+
+fun buildCreditCardRoute(
+  paymentMethodId: String,
+  isPreSelected: Boolean = false,
+) =
+  "$CREDIT_CARD_ROUTE?$CREDIT_CARD_PAYMENT_ID_ARG=${paymentMethodId}&$IS_PRE_SELECTED=${isPreSelected}"
+
+fun creditCardPaymentScreen(onFinish: (Boolean) -> Unit) = ScreenData.withAnalytics(
+  route = CREDIT_CARD_FULL_ROUTE,
+  screenAnalyticsName = "CreditCard",
+  arguments = creditCardPaymentArguments
+) { args, _, popBackStack ->
+  val paymentMethodId = args?.getString(CREDIT_CARD_PAYMENT_ID_ARG)!!
+  val isPreSelected = args.getBoolean(IS_PRE_SELECTED)
+  BuildAdyenCreditCardScreen(
+    onFinish = onFinish,
+    popBackStack = popBackStack,
+    paymentMethodId = paymentMethodId,
+  )
+
+  BackHandler(enabled = isPreSelected) {
+    onFinish(false)
+  }
+}
+
+@PreviewLight
+@Composable
+private fun AdyenCreditCardScreenPortraitPreview() {
+  AptoideTheme {
+    AppGamesPaymentBottomSheet {
+      AdyenCreditCardScreen(
+        packageName = "packageName",
+        onBuyClick = {},
+        onOtherPaymentMethodsClick = {},
+        onBuyClickEnabled = Random.nextBoolean(),
+        paymentView = {
+          Box(
+            modifier = Modifier
+              .padding(top = 16.dp)
+              .height(150.dp)
+              .fillMaxWidth()
+              .border(1.dp, Palette.Black),
+            contentAlignment = Alignment.Center
+          ) {
+            Text(text = "Adyen View")
+          }
+        }
+      )
+    }
+  }
+}
+
+@PreviewLandscapeLight
+@Composable
+private fun AdyenCreditCardScreenLandscapePreview() {
+  AptoideTheme {
+    AppGamesPaymentBottomSheet {
+      AdyenCreditCardScreen(
+        packageName = "packageName",
+        onBuyClick = {},
+        onOtherPaymentMethodsClick = {},
+        onBuyClickEnabled = Random.nextBoolean(),
+        paymentView = {
+          Box(
+            modifier = Modifier
+              .padding(top = 16.dp)
+              .height(150.dp)
+              .fillMaxWidth()
+              .border(1.dp, Palette.Black),
+            contentAlignment = Alignment.Center
+          ) {
+            Text(text = "Adyen View")
+          }
+        }
+      )
+    }
+  }
+}
+
+@Composable
+private fun BuildAdyenCreditCardScreen(
+  onFinish: (Boolean) -> Unit,
+  popBackStack: () -> Unit,
+  paymentMethodId: String,
+) {
+  val context = LocalContext.current as ComponentActivity
+  val lifecycleOwner = LocalLifecycleOwner.current
+  val (uiState, buy) = rememberAdyenCreditCardUIState(paymentMethodId)
+  var onBuyClick by remember { mutableStateOf<(() -> Unit)?>(null) }
+
+  var finished by remember { mutableStateOf(false) }
+
+  AdyenCreditCardStateEffect(paymentMethodId, uiState)
+
+  AppGamesPaymentBottomSheet(
```

Comment:
this bottom sheet invocation should be included in a single method. it would have two advantages:
1 - you could remove the bottom sheet from the previews
2 - you can create a state in the preview to preview all states